### PR TITLE
Revise Columns, add meter, add fractional drawing, optimize palettes and setpixel, remove green temp effect

### PR DESCRIPTION
**Pull request by davepl**
Adds a meter to the monitor, revises columns.
Adds fractional pixel drawing (double precision and single)
Optimized palette code
Optimized SetPixelsF code
Removed the greenfilleffect and its file
Refactors monitor to move code around between files

### CODE REVIEW

File: palette.h [hunk 1]
```diff
@@ -8,90 +8,183 @@
 #include "pixeltypes.h"
 #include "palette.h"
 
+template <size_t N>
 class Palette
 {
 protected:
-    std::vector<CRGB> colorEntries;
+    std::array<CRGB, N> _colorEntries;
```
Comment (rbergen) [inline palette.h:15]: You can probably drop the std:: here.

File: palette.cpp [hunk 2]
```diff
@@ -5,26 +5,34 @@
...
-    CRGB(0, 255, 255), // Cyan
-    CRGB(0, 0, 255),   // Blue
-    CRGB(128, 0, 128)  // Purple
-};
+// Rainbow - 8 colors for smoother transitions in a performant power of 2 size
 
-const std::vector<CRGB> Palette::ChristmasLights = 
-{
+template<>
+const Palette<8> Palette<8>::Rainbow({
```
Comment (rbergen) [inline palette.cpp:11]: These static `Palette<N>` class member variable palettes are not used anywhere; the `PaletteEffect<N>` instances created in main.cpp use the palettes in the `StandardPalettes` namespace as declared in pixeltypes.h. I think one set of standard palettes should stay, and the other should go.

**Review by rbergen — COMMENTED** [review]
Just a few small things, although I think a decision about which set of "standard palettes" to retain (and use) should be made.

I note that I still have to test the reconnectCount display in ledmon; I'll do that tomorrow.

**Comment by davepl** [comment]
I’m starting to regret using a template parameter for Palette… I suspect the code that gets generated is a lot more efficient, but you wind up with a diffierent type for each size of palette, which gets complicated in the persistence case.

- Dave

**Comment by rbergen** [comment]
Yeah, I was thinking about that, but I assumed you opted for the compile-time optimization you get out of it. It is also annoying C++ can't deduce the value for a size_t template parameter, when it can deduce typename parameters.

I wouldn't hold it against you if you went back to using vectors for Palettes, if that's what you're wondering.

**Comment by rbergen** [comment]
Closing this in favor of #10.

**Pull request closed without merging**